src/components/Calculator.js

```diff
@@ -1,20 +1,51 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './Calculator.css';
+import calculate from '../logic/calculate';
 
 function Calculator() {
-  const handleClick = (value) => {
-    console.log(value);
+  const [display, setDisplay] = useState('0');
+  const [total, setTotal] = useState(null);
+  const [next, setNext] = useState(null);
+  const [operation, setOperation] = useState(null);
+
+  const handleClick = (buttonName) => {
+    if (buttonName === 'AC') {
+      setTotal(null);
+      setNext(null);
+      setOperation(null);
+      setDisplay('0');
+      return;
+    }
+    const result = calculate({ total, next, operation }, buttonName);
+    if (result) {
+      setTotal(result.total);
+      setNext(result.next);
+      setOperation(result.operation);
+      setDisplay((display === '0' && buttonName !== '.') ? buttonName : display + buttonName);
+    }
+    if (buttonName === '=') {
+      setDisplay(result.total);
+      setTotal(result.total);
+      setNext(null);
+      setOperation(null);
+    }
   };
 
   return (
     <div className="calculator">
-      <div className="screen" />
+      <div className="screen">{display}</div>
       <div className="buttons">
+        <div className="row">
+          <button className="button" type="button" onClick={() => handleClick('AC')}>AC</button>
+          <button className="button" type="button" onClick={() => handleClick('+/-')}>+/-</button>
+          <button className="button" type="button" onClick={() => handleClick('%')}>%</button>
+          <button className="button operator" type="button" onClick={() => handleClick('÷')}>÷</button>
```

Comment:
- [ ]  You have done an amazing job for building a well functioning app, but one thing missing in the use of props.  You should have a specific child component like button, etc, so that you can pass props from parent component to child component correctly.